### PR TITLE
fsttest: discard header changes after send

### PR DESCRIPTION
**Pull request by fgsch**
This should correctly ignore headers set too late to make it into the response.

### CODE REVIEW

File: fsttest/recorder.go [hunk 1]
```diff
@@ -28,12 +29,18 @@ func NewRecorder() *ResponseRecorder {
 
 // Header returns the response headers to mutate within a handler.
 func (r *ResponseRecorder) Header() fsthttp.Header {
-	return r.HeaderMap
+	if !r.headersDone {
+		return r.HeaderMap
+	}
+	return r.HeaderMap.Clone()
```
Comment (joeshaw) [inline fsttest/recorder.go:35]: Do you mind adding a comment here to explain the purpose?  I'm not sure it's obvious without additional context.

**Review by joeshaw — COMMENTED** [review]
LGTM but I have a request for a comment explaining why we return a clone.

File: fsttest/recorder.go (hunk 1 above)
Comment (cee-dub) [reply fsttest/recorder.go:35]: This is a good point and I agree. Being part of our test harness, I was initally being more liberal about the lack of comment.

Comment (fgsch) [reply fsttest/recorder.go:35]: Sure, I will update the PR.

Comment (fgsch) [reply fsttest/recorder.go]: Done. Let me know if that works for you.

File: fsttest/recorder.go [hunk 2]
```diff
@@ -28,12 +29,20 @@ func NewRecorder() *ResponseRecorder {
 
 // Header returns the response headers to mutate within a handler.
 func (r *ResponseRecorder) Header() fsthttp.Header {
-	return r.HeaderMap
+	if !r.headersDone {
+		return r.HeaderMap
+	}
+	// Once the send the headers, return a copy so any changes
+	// are discarded.
+	return r.HeaderMap.Clone()
```
Comment (cee-dub) [inline fsttest/recorder.go:37]:
```suggestion
	if !r.headersSent {
		return r.HeaderMap
	}
	// Once sent, return a copy so any changes are discarded.
	return r.HeaderMap.Clone()
```
I find the name `headersSent` to be more self-explanatory than `headersDone`, so I suggest changing that throughout.

**Review by cee-dub — COMMENTED** [review]
Could have given better feedback the first time through. Apologies for the back and forth!

File: fsttest/recorder.go (hunk 2 above)
Comment (fgsch) [reply fsttest/recorder.go]: Done.

**Comment by fgsch** [comment]
Updated to address feedback. PTAL

**Review by cee-dub — APPROVED** [review]
LGTM!

**Pull request merged**